core/src/test/java/io/reacted/core/drivers/local/SystemLocalDriversTest.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    private static Stream<Arguments> logger() {
+        return Stream.of(
+                Arguments.of(true, "SIMPLIFIED_LOGGING_DIRECT_COMMUNICATION-"),
+                Arguments.of(false, "LOGGING_DIRECT_COMMUNICATION-"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("logger")
+    void fileCreatedInDirectCommunicationLogger(boolean isSimplifiedLogger, String channelId) {
+        var reActorSystem = gerReactorSystem(isSimplifiedLogger, TMP_TEST_LOGS_TEST_DIRECT_COMMUNICATION_TXT);
+        Assertions.assertEquals("DIRECT_COMMUNICATION@" + channelId,
+                                reActorSystem.getLoopback().getBackingDriver().getChannelId().toString());
+        Assertions.assertTrue(new File(TMP_TEST_LOGS_TEST_DIRECT_COMMUNICATION_TXT).exists());
+        reActorSystem.shutDown();
+    }
+
+    private static Stream<Arguments> incorrectLogPaths() {
+        return Stream.of(
+                // no file name specified
+                Arguments.of("/tmps/"),
+                // directory not existent
+                Arguments.of("/tmps/test.txt"));
+    }
+
+    @ParameterizedTest
+    @MethodSource("incorrectLogPaths")
+    void logFileNotCreatedWhenFileNameIsIncorrect(String logPath) {
+        Assertions.assertThrows(UncheckedIOException.class, () -> gerReactorSystem(true, logPath));
+    }
+
+    private static Stream<Arguments> logPaths() {
+        return Stream.of(
+                Arguments.of(true, TMP_TEST_LOGS_TEST_DIRECT_COMMUNICATION_TXT),
+                Arguments.of(false, "/tmp/test_logs/testDirect Communication.txt"),
+                Arguments.of(false, "/tmp/test_logs/" + Instant.now().toEpochMilli() + ".txt")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("logPaths")
+    void messageIsLoggedInDirectCommunicationLogger(boolean isSimplifiedLogger, String logPath) {
+        var reActorSystem = gerReactorSystem(isSimplifiedLogger, logPath);
+
+        await().until(() -> Files.readString(Path.of(logPath), StandardCharsets.US_ASCII).strip(),
+                      Matchers.containsString("Init"));
+        reActorSystem.shutDown();
+    }
+
+    private static Stream<Arguments> loggerType() {
+        return Stream.of(
+                Arguments.of(true),
+                Arguments.of(false)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("loggerType")
+    void deliveryStatusIsDeliveredWhenMessageIsSent(boolean isSimplifiedLogger) {
+        var reActorSystem = gerReactorSystem(isSimplifiedLogger, TMP_TEST_LOGS_TEST_DIRECT_COMMUNICATION_TXT);
+        var deliveryAttempt = reActorSystem.getSystemSink().tell("Payload of this message")
+                .toCompletableFuture()
+                .join();
+        reActorSystem.shutDown();
+        deliveryAttempt.filter(DeliveryStatus::isDelivered)
+                .ifError(Assertions::fail);
+    }
+
+    private static ReActorSystem gerReactorSystem(boolean isSimplifiedLogger, String logFilePath) {
```

Comment:
It would be more future proof providing the local driver as a parameter and creating different named overloads instead of using a boolean

Reply:
we can discuss it